rs/index/src/multi_spann/index.rs

```diff
@@ -54,9 +54,12 @@ impl<Q: Quantizer> MultiSpannIndex<Q> {
             ivf_type,
         };
 
-        // Iterate over invalidated IDs and call invalidate for each entry
+        // Iterate over invalidated ids and invalidate each entry. Do not call
+        // MultiSpannIndex::invalidate directly as this also appends invalidated ids to the
+        // storage, whereas these ids are already stored in the storage.
         for invalidated_id in invalidated_ids.iter() {
-            index.invalidate(invalidated_id.user_id, invalidated_id.doc_id)?;
+            let spann_index = index.get_or_create_index(invalidated_id.user_id)?;
+            let _ = spann_index.invalidate(invalidated_id.doc_id);
```

Comment:
you don't need to create `invalidated_ids`. All you need to do is:
```
let invalided_ids = index.invalidated_ids_storage.read();
for id in invalidated_ids.iter() {

}
```
Also, change the `.iter` function `&self` instead of `&mut self`